lib/landrush/win_network_config.rb

```diff
@@ -0,0 +1,159 @@
+require 'ipaddr'
+
+# This class configures the network interface on Windows for use with the Landrush DNS server.
+# It makes use of the netsh executable which is assumed to be available on the Windows host.
+module Landrush
+  class WinNetworkConfig
+    # Windows registry path under which network interface configuration is stored
+    INTERFACES = 'SYSTEM\\CurrentControlSet\\Services\\Tcpip\\Parameters\\Interfaces'.freeze
+
+    def initialize(env={})
+      @env = env
+    end
+
+    # Checks that all required tools are on the PATH and that the Wired AutoConfig service is started
+    def check_prerequisites
+      info('Checking prerequisites for configuring DNS on host')
+
+      if self.class.which('netsh').nil?
+        info('Cannot find \'netsh\' on the PATH. Unable to configure DNS on host. Try manual configuration.')
+        return false
+      end
+
+      if self.class.which('net').nil?
+        info('Cannot find \'net\' on the PATH. Unable to configure DNS on host. Try manual configuration.')
+        return false
+      end
+
+      if `net start`.match(/Wired AutoConfig/m).nil?
+        info('Starting \'Wired AutoConfig\' service')
+        if self.class.admin_mode?
+          `net start dot3svc`
+        else
+          require 'win32ole'
+          shell = WIN32OLE.new('Shell.Application')
+          shell.ShellExecute('net', 'start dot3svc', nil, 'runas', 1)
+        end
+        service_started = `net start`.match(/Wired AutoConfig/m).nil?
+        unless service_started
+          info('Unable to \'Wired AutoConfig\' service. Unable to configure DNS on host. Try manual configuration.')
+          return false
+        end
+      end
+      true
+    end
+
+    # Does the actual update of the network configuration
+    def update_network_adapter(ip, name_server, domain)
+      # Need to defer loading to ensure cross OS compatibility
+      require 'win32/registry'
+      ensure_admin_privileges(__FILE__.to_s, ip, name_server, domain)
+      network_name = get_network_name(ip)
+      info("Setting DNS server for network \'#{network_name}\' to #{ip} and search domain to \'#{domain}\'")
+      network_guid = get_guid(network_name)
+      interface_path = INTERFACES + "\\{#{network_guid}}"
+      Win32::Registry::HKEY_LOCAL_MACHINE.open(interface_path, Win32::Registry::KEY_ALL_ACCESS) do |reg|
+        reg['NameServer'] = name_server
+        reg['Domain'] = domain
+      end
+    end
+
+    # Cross-platform way of finding an executable in the $PATH.
+    def self.which(cmd)
```

Comment:
You may want to use vagrant supported `Which` module https://github.com/mitchellh/vagrant/blob/master/lib/vagrant/util/which.rb

Somehow it uses similar code as you are using :smile: